plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -717,18 +773,25 @@ function MosaicMenuItem:paintTo(bb, x, y)
     end
 
     -- to which we paint over a dogear if needed
-    if corner_mark and self.do_hint_opened and self.been_opened then
-        -- align it on bottom right corner of sub-widget
-        local target =  self[1][1][1]
-        local ix
-        if BD.mirroredUILayout() then
-            ix = math.floor((self.width - target.dimen.w)/2)
-        else
-            ix = self.width - math.ceil((self.width - target.dimen.w)/2) - corner_mark:getSize().w
-        end
-        local iy = self.height - math.ceil((self.height - target.dimen.h)/2) - corner_mark:getSize().h
-        -- math.ceil() makes it looks better than math.floor()
-        corner_mark:paintTo(bb, x+ix, y+iy)
+    -- if corner_mark and self.do_hint_opened and self.been_opened then
+    --     -- align it on bottom right corner of sub-widget
+    --     local target =  self[1][1][1]
+    --     local ix
+    --     if BD.mirroredUILayout() then
+    --         ix = math.floor((self.width - target.dimen.w)/2)
+    --     else
+    --         ix = self.width - math.ceil((self.width - target.dimen.w)/2) - corner_mark:getSize().w
+    --     end
+    --     local iy = self.height - math.ceil((self.height - target.dimen.h)/2) - corner_mark:getSize().h
+    --     -- math.ceil() makes it looks better than math.floor()
+    --     corner_mark:paintTo(bb, x+ix, y+iy)
+    -- end
+
+
+    if self.progress ~= nil then
+        self.progress.width = self[1][1][1].width * 0.66
+        local pos_x = x+self.width - self.progress.width - ((self.width - self[1][1][1].width) / 2) -self.progress.bordersize
+        self.progress:paintTo(bb, pos_x, y+self.height-self.progress.height-self.progress.bordersize)
```

Comment:
I would also not create and store a ProgressWidget in each MosaicMenuItem.
I would, in :update(), just store the % value.
And here in `MosaicMenuItem:paintTo()`, either:
- instantiate the ProgressWidget here, paint it, free() it and forget it
- or just instantiate a single ProgressWidget at module level, like it is done for `corner_mark = IconWidget:new{}`, and just here `:setPercentage(this book %)` and paint it as this x/y